x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleAction.java

```diff
@@ -138,22 +167,16 @@ public List<Step> toSteps(Client client, String phase, StepKey nextStepKey) {
         // Mark source index as read-only
         ReadOnlyStep readOnlyStep = new ReadOnlyStep(readOnlyKey, cleanupDownsampleIndexKey, client);
 
-        // We generate a unique downsample index name, but we also retry if the allocation of the downsample index
-        // is not possible, so we want to delete the "previously generated" downsample index (this is a no-op if it's
-        // the first run of the action, and we haven't generated a downsample index name)
-        CleanupTargetIndexStep cleanupDownsampleIndexStep = new CleanupTargetIndexStep(
-            cleanupDownsampleIndexKey,
-            generateDownsampleIndexNameKey,
-            client,
-            (indexMetadata) -> IndexMetadata.INDEX_DOWNSAMPLE_SOURCE_NAME.get(indexMetadata.getSettings()),
-            (indexMetadata) -> indexMetadata.getLifecycleExecutionState().downsampleIndexName()
-        );
+        // Before the downsample action was retry-able, we used to generate a unique downsample index name and delete the previous index in
+        // case a failure occurred. The downsample action can now retry execution in case of failure and start where it left off.
+        NoopStep cleanupDownsampleIndexStep = new NoopStep(cleanupDownsampleIndexKey, generateDownsampleIndexNameKey);
 
-        // Generate a unique downsample index name and store it in the ILM execution state
-        GenerateUniqueIndexNameStep generateDownsampleIndexNameStep = new GenerateUniqueIndexNameStep(
+        // Generate a predictable downsample index name and store it in the ILM execution state
+        GenerateUniqueIndexNameStep generateDownsampleIndexNameStep = new GenerateDownsampleIndexNameStep(
```

Comment:
If the name is stable I don't think we need to store it in the cluster state (and `GenerateDownsampleIndexNameStep` doesn't need to exist anymore). 

`Downsample` step has the fixed interval so could figure out the name of the target downsample index itself I believe?